### PR TITLE
feat(DEV-12593): Merge create/preview steps of invoice creation into a single flow

### PR DESCRIPTION
**Pull request by chebum**
This PR opens invoice for editing immediately after its creation. Previously the user were directed back to the invoices list.

The PR contains a long test that fills required fields when creating a new Receivable. I didn't mock nested components because we had a broken test that tested Receivable creation. Therefore my longer test both restores the disabled test from 2022 and tests the new receivable creation flow.

### CODE REVIEW

**Comment by chebum** [comment]
The PR contains a long test that fills required fields when creating a  new Receivable. I didn't mock nested components because we had a broken test that tested Receivable creation. Therefore my longer test both restores the disabled test from 2022 and tests the new receivable creation flow.

File: packages/sdk-react/src/utils/test-utils.tsx [hunk 1]
```diff
@@ -211,6 +212,34 @@ export async function waitUntilTableIsLoaded(
...
+ *
+ * @param predicate Predicate checking for condition
+ * @param timeout Wait function timeout
+ */
+export function waitForCondition(
+  predicate: () => boolean,
+  timeout: number = 1_000
+) {
+  return new Promise<void>((resolve, reject) => {
+    const interval = 50;
```
Comment (vkuprin) [inline packages/sdk-react/src/utils/test-utils.tsx:226]: Can we have this as passed argument with default value 50?

File: packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/CreateReceivables.test.tsx [hunk 2]
```diff
@@ -44,37 +57,100 @@ describe('CreateReceivables', () => {
...
+      timeout: 3_000,
     });
-  });
+
+    // Create invoice
+    const nextPageButton = await nextPageButtonPromise;
+    fireEvent.click(nextPageButton);
+
+    // Compose email dialog opens
+    await waitForCondition(
```
Comment (vkuprin) [inline packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/CreateReceivables.test.tsx:149]: Can we please make some cleanups here and remove // comments as code does looks as self explanatory

Comment (chebum) [reply packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/CreateReceivables.test.tsx]: I removed comments as suggested

File: packages/sdk-react/src/utils/test-utils.tsx (hunk 1 above)
Comment (chebum) [reply packages/sdk-react/src/utils/test-utils.tsx]: Passed the check interval as a function parameter

**Review by vkuprin — APPROVED** [review]

**Pull request merged**